packages/core/src/browser/status-bar/status-bar.tsx

```diff
@@ -62,14 +62,21 @@ export const StatusBar = Symbol('StatusBar');
 
 export interface StatusBar {
     setBackgroundColor(color?: string): Promise<void>;
+    setClassName(state: StatusBarState): Promise<void>;
```

Comment:
status bar should not care about it

Reply:
the issue with `setForegroundColor` or `setColor` is that it works initially but if a file is selected for instance after being offline it will be rendered with a white font.

Reply:
I tried many different combinations on what to do but the easiest sure way to do it for me was to update the status-bar class name when offline or online

Reply:
> the issue with setForegroundColor or setColor is that it works initially but if a file is selected for instance after being offline it will be rendered with a white font.

What is the cause of its to be rendered with a white font after being offline?

Reply:
We re-render some status-bar items such as `line, column` if a new file is clicked while offline. Since our default css uses a white font color I believe it's using that color when rendering. Calling the `handleOffline()` is done once when the state is changed, so any updated to a status-bar item will not get the correct styling.

Reply:
> Since our default css uses a white font color I believe it's using that color when rendering.

How about making sure that is using the last set foreground color and only if it is not set (`null`) use the default color?

Reply:
Making `StatusBar` suitable only for a particular client (`ApplicationConnectionStatusContribution`)  is not a good API.

Reply:
thanks @akosyakov I'll make the changes